mlflow/store/sqlalchemy_store.py

```diff
@@ -0,0 +1,222 @@
+import sqlalchemy
+from sqlalchemy import orm
+from sqlalchemy.exc import IntegrityError
+from mlflow.store.dbmodels import models
+from mlflow import entities
+from mlflow.store.abstract_store import AbstractStore
+from mlflow.entities import ViewType
+from mlflow.exceptions import MlflowException
+import mlflow.protos.databricks_pb2 as error_codes
+
+
+class SqlAlchemyStore(AbstractStore):
+
+    def __init__(self, db_uri):
+        super(SqlAlchemyStore, self).__init__()
+        self.engine = sqlalchemy.create_engine(db_uri)
+        models.Base.metadata.create_all(self.engine)
+        models.Base.metadata.bind = self.engine
+        db_session = orm.sessionmaker(bind=self.engine)
+        self.session = db_session()
+
+    def _save_to_db(self, objs):
+        """
+        Store in db
+        """
+
+        if type(objs) is list:
+            self.session.add_all(objs)
+        else:
+            # single object
+            self.session.add(objs)
+
+        self.session.commit()
+
+    def _get_or_create(self, model, **kwargs):
+        instance = self.session.query(model).filter_by(**kwargs).first()
+        create = False
+
+        if instance:
+            return instance, create
+        else:
+            instance = model(**kwargs)
+            self._save_to_db(instance)
+            create = True
+
+        return instance, create
+
+    def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
+        mappings = {
+            ViewType.ACTIVE_ONLY: entities.Experiment.ACTIVE_LIFECYCLE,
+            ViewType.DELETED_ONLY: entities.Experiment.DELETED_LIFECYCLE
+        }
+
+        lifecycle_stage = mappings[view_type]
+        experiments = []
+        for exp in self.session.query(models.SqlExperiment).filter_by(
+                lifecycle_stage=lifecycle_stage, is_deleted=False):
```

Comment:
It is a bit confusing to have both `lifecycle_stage=DELETED` and `is_delete` boolean. How would that work? Can we use just `is_deleted`?

Looks like `delete_experiment` and `restore_experiment`, below -- both update `is_deleted` flag only.